### ISSUE TITLE
Rewrite the services/flash-messages.js service

### ISSUE THREAD

**Issue by izelnakri** [poteto/ember-cli-flash#271]
Hi, I'm currently trying to use this addon with new ember-resolver(for module unification), in my new `mber` project. It was all going good until I run my tests. They were all passing before my `mber` migration but this time `only some` of my acceptance tests get `flashMessages.danger method  is not a function`, `flashMessages.success method  is not a function` errors. In some other tests these methods are called correctly.

So I started digging around the source code and saw that there is an internal `this.registerTypes()` call on each init:
https://github.com/poteto/ember-cli-flash/blob/master/addon/services/flash-messages.js#L125

First of all I have to say this service code is very poorly written, it is very hard to follow along and I have read many messy source-code before:

I dont know where that `flashMessageDefaults` comes from, its not explicit in the source code, apparently `this` context is shared with something(possibly environment.js?) implicitly. Surely there is a definition in the `environment.js` and it gets injected to the Application in my tests entrypoint(`tests/test-helper`), but that is not the service context. Most of the code is 3 years old, surely it could be written better today, with less implicitness, I see many unit tests already for service.

I will keep debugging to see how I lose that implicitly injected `flashMessageDefaults ` in some of the acceptance tests. It could also be possible that owner.registration and owner.lookups changes/restrictions in new ember-resolver causes this(there are some registrations in the initializers). Meanwhile, I highly suggest the maintainer to rewrite that service module once more.

**Comment by izelnakri** [comment]
The issue was probably in the ember version or the resolver Im using. I did the following two things and everything is working:

in my ember modules
```js
export default Component.extend({
  flashMessages: service()
})
```
to:
```js
export default Component.extend({
  flashMessages: service('flash-messages')
})
```

in test owner.lookups:
context.owner.lookup('service:flashMessages') to context.owner.lookup('service:flash-messages')

**Comment by olleolleolle** [comment]
@izelnakri What do you think? Is this issue closable with the changes now in place?

**Comment by sbatson5** [comment]
Closing. @izelnakri can bug me on our work slack if this is still an issue 😜
